analyzers/tests/SonarAnalyzer.UnitTest/Rules/TestClassShouldHaveTestMethodTest.cs

```diff
@@ -31,32 +31,43 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class TestClassShouldHaveTestMethodTest
     {
+        private readonly VerifierBuilder verifierBuilder = new VerifierBuilder<TestClassShouldHaveTestMethod>();
+
         [DataTestMethod]
         [DataRow("2.5.7.10213")]
         [DataRow(Constants.NuGetLatestVersion)]
         public void TestClassShouldHaveTestMethod_NUnit(string testFwkVersion) =>
-            OldVerifier.VerifyAnalyzer(
-                @"TestCases\TestClassShouldHaveTestMethod.NUnit.cs",
-                new TestClassShouldHaveTestMethod(),
-                NuGetMetadataReference.NUnit(testFwkVersion));
+            verifierBuilder
+                .AddPaths(@"TestClassShouldHaveTestMethod.NUnit.cs")
+                .AddReferences(NuGetMetadataReference.NUnit(testFwkVersion))
+                .Verify();
+
+        [DataTestMethod]
+        [DataRow("3.0.0")]
+        [DataRow(Constants.NuGetLatestVersion)]
+        public void TestClassShouldHaveTestMethod_NUnit3(string testFwkVersion) =>
+            verifierBuilder
+                .AddPaths(@"TestClassShouldHaveTestMethod.NUnit3.cs")
+                .AddReferences(NuGetMetadataReference.NUnit(testFwkVersion))
+                .Verify();
 
         [DataTestMethod]
         [DataRow("1.1.11")]
         [DataRow(Constants.NuGetLatestVersion)]
         public void TestClassShouldHaveTestMethod_MSTest(string testFwkVersion) =>
-            OldVerifier.VerifyAnalyzer(
-                @"TestCases\TestClassShouldHaveTestMethod.MsTest.cs",
-                new TestClassShouldHaveTestMethod(),
-                NuGetMetadataReference.MSTestTestFramework(testFwkVersion));
+            verifierBuilder
+                .AddPaths(@"TestClassShouldHaveTestMethod.MsTest.cs")
+                .AddReferences(NuGetMetadataReference.MSTestTestFramework(testFwkVersion))
+                .Verify();
 
 #if NET
         [DataTestMethod]
         public void TestClassShouldHaveTestMethod_CSharp9() =>
-            OldVerifier.VerifyAnalyzerFromCSharp9Library(
-                @"TestCases\TestClassShouldHaveTestMethod.CSharp9.cs",
-                new TestClassShouldHaveTestMethod(),
-                NuGetMetadataReference.MSTestTestFramework(Constants.NuGetLatestVersion)
-                    .Concat(NuGetMetadataReference.NUnit(Constants.NuGetLatestVersion)));
+            verifierBuilder
```

Comment:
This one had `VerifyAnalyzerFromCSharp9Library` so it will need `.WithOptions(ParseOptionsHelper.FromCSharp9)` to work.
This UT should fail now.